src/DurableTask.Core/TaskActivityDispatcher.cs

```diff
@@ -137,7 +141,7 @@ async Task OnProcessWorkItemAsync(TaskActivityWorkItem workItem)
                 }
 
                 this.logHelper.TaskActivityStarting(orchestrationInstance, scheduledEvent);
-                TaskActivity? taskActivity = this.objectManager.GetObject(scheduledEvent.Name, scheduledEvent.Version);
+                TaskActivity? taskActivity = this.objectManager.GetObject(scheduledEvent.Name, scheduledEvent.Version, this.dataConverter);
```

Comment:
We can avoid having to change object manager contract by setting the data converter here ourselves:
```suggestion
                TaskActivity? taskActivity = this.objectManager.GetObject(scheduledEvent.Name, scheduledEvent.Version);
                if (taskActivity is IDataConverterConfigurable configurable)
                {
                    configurable.DataConverter = this.dataConverter;
                }
```